utils/activedirectory2_functions.go

```diff
@@ -48,7 +48,7 @@ func ActiveDirectory2GetFunc(ctx context.Context, _client interface{}, attr map[
 		return response, err
 	}
 
-	err = UnmarshelBodyToMap(response, &temp)
```

Comment:
I think in go one line syntax is more preffered:
`
if err = UnmarshalBodyToMap(response, &temp); err != nil {
    return response, err
}

`